egui_extras/src/table.rs

```diff
@@ -351,7 +351,65 @@ pub struct TableBody<'a> {
     end_y: f32,
 }
 
+pub trait TableRowBuilder {
+    fn row_heights(&self, widths: &Vec<f32>) -> Box<dyn Iterator<Item = f32> + '_>;
+    fn populate_row(&self, index: usize, row: TableRow<'_, '_>);
+}
+
 impl<'a> TableBody<'a> {
+    pub fn heterogenous_rows(mut self, builder: impl TableRowBuilder) {
+        let mut heights = builder.row_heights(&self.widths);
+
+        let max_height = self.end_y - self.start_y;
+        let delta = self.start_y - self.layout.current_y();
+
+        // cumulative height of all rows above those being displayed
+        let mut height_above_visible = 0.0;
+        // cumulative height of all rows below those being displayed
+        let mut height_below_visible = 0.0;
+
+        let mut row_index = 0;
+
+        let mut above_buffer_set = false;
+        let mut current_height = 0.0; // used to track height of visible rows
+        while let Some(height) = heights.next() {
+            // when delta is greater than height above 0, we need to increment the row index and
+            // update the height above visble with the current height then continue
+            if height_above_visible < delta {
+                row_index += 1;
+                height_above_visible += height;
```

Comment:
Accumulating `f32` will lead to precision issues really quickly (even worse than those [already present](https://github.com/emilk/egui/issues/1391)). Better to use `f64` for `height_above_visible`,  `height_below_visible` and `current_height`.

Reply:
Should the `height` values be given in `f64` then also? I don't know much about precision issues between different bit widths. But I did go ahead and make this change.

Reply:
No, `height` can be `f32`. It is the accumulator that needs to be wide.